infrastructure/create_pipeline.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+echo Assign variables
+for ARGUMENT in "$@"
+do
+   KEY=$(echo $ARGUMENT | cut -f1 -d=)
+
+   KEY_LENGTH=${#KEY}
+   VALUE="${ARGUMENT:$KEY_LENGTH+1}"
+
+   export "$KEY"="$VALUE"
+done
+
+echo Ensure all necessary variables have been declared
+if [[ -z $PIPELINE_DATE || -z $AWS_PROFILE || -z $BUILDKITE_COMMIT ]]; then
+  echo "ERROR: PIPELINE_DATE, BUILDKITE_COMMIT and AWS_PROFILE are mandatory arguments.";
+  exit 1;
+fi
+
+echo Create zips for packages
+yarn workspace @weco/content-pipeline run package && yarn workspace @weco/content-unpublisher run package
+
+echo Upload zips to S3
+bash ../.buildkite/scripts/upload_lambda_package.sh content-pipeline-$PIPELINE_DATE ../pipeline/package.zip $AWS_PROFILE=$AWS_PROFILE $BUILDKITE_COMMIT=$BUILDKITE_COMMIT
+bash ../.buildkite/scripts/upload_lambda_package.sh content-unpublisher-$PIPELINE_DATE ../unpublisher/package.zip $AWS_PROFILE=$AWS_PROFILE $BUILDKITE_COMMIT=$BUILDKITE_COMMIT
```

Comment:
The shebang (`#!/usr/bin/env bash`) at the top of the script tells the shell how to execute this script if the script itself is marked as executable. So you don't need to reference `bash` here if those scripts have a shebang.

I'm not sure setting the $AWS_PROFILE=$AWS_PROFILE is working as expected. I think you may want to export this environment variable in the script before executing the other scripts so that it is available in the process of the child scripts. This is _already done_ when you set `AWS_PROFILE=catalogue_developer` above.

`$BUILDKITE_COMMIT=$BUILDKITE_COMMIT` is not used by the consuming script AFAICT and can be omitted.

It's also a good idea to use absolute paths by some known reference if possible to avoid things not being where you expect if the script is run from an unexpected location, so we use `ROOT` as defined above.

```suggestion
$ROOT/.buildkite/scripts/upload_lambda_package.sh \
  content-pipeline-$PIPELINE_DATE $ROOT/pipeline/package.zip
$ROOT/.buildkite/scripts/upload_lambda_package.sh \
  content-unpublisher-$PIPELINE_DATE $ROOT/unpublisher/package.zip 
```

Reply:
BUILDKITE COMMIT is used here, yes? https://github.com/wellcomecollection/content-api/blob/main/.buildkite/scripts/upload_lambda_package.sh#L10

Reply:
Shall I just make it `dev`, is there a reason we'd want a different one?

Reply:
Spoke in person, needed to export the value and has been set to dev explicitely